Amplitude/Amplitude.m

```diff
@@ -533,6 +531,20 @@ - (void)logEvent:(NSString*) eventType withEventProperties:(NSDictionary*) event
     }];
 }
 
+- (void)truncateEventQueues
+{
+    AMPDatabaseHelper *dbHelper = [AMPDatabaseHelper getDatabaseHelper];
+    int numEventsToRemove = self.eventMaxCount > kAMPEventRemoveBatchSize ? kAMPEventRemoveBatchSize : self.eventMaxCount/2;
```

Comment:
I think this should be more like min(max(1, self.eventMaxCount / 10), kAMPEventRemoveBatchSize)